openrefine.yaml

```diff
@@ -0,0 +1,61 @@
+package:
+  name: openrefine
+  version: 3.7.5
+  epoch: 0
+  description: "OpenRefine is a free, open source power tool for working with messy data and improving it"
+  copyright:
+    - license: BSD-3-Clause
+  dependencies:
+    runtime:
+      - openjdk-20-jre
+      - bash
+      - wget
+
+environment:
+  contents:
+    packages:
+      - wolfi-base
+      - build-base
+      - bash
+      - openjdk-20-jre
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/OpenRefine/OpenRefine
+      tag: ${{package.version}}
+      expected-commit: a04fb5fbd0d35d7b727bdca7d34bbca8245427d8
+
+  - name: Set directories
+    runs: |
+      mkdir -p "${{targets.destdir}}"/usr/bin
+      mkdir -p "${{targets.destdir}}"/usr/share/${{package.name}}/
+      mkdir -p "${{targets.destdir}}"/usr/share/licenses/${{package.name}}/
+
+  - name: Fix bash path for /usr/share/openrefine/refine
+    runs: |
+      sed -i '1s|^#!/usr/bin/env bash$|#!/bin/bash|' refine
+
+  - name: Install
+    runs: |
+      mv benchmark "${{targets.destdir}}"/usr/share/${{package.name}}/
+      mv conf "${{targets.destdir}}"/usr/share/${{package.name}}/
+      mv extensions "${{targets.destdir}}"/usr/share/${{package.name}}/
+      mv graphics "${{targets.destdir}}"/usr/share/${{package.name}}/
+      mv IDEs "${{targets.destdir}}"/usr/share/${{package.name}}/
+      mv main "${{targets.destdir}}"/usr/share/${{package.name}}/
+      mv packaging "${{targets.destdir}}"/usr/share/${{package.name}}/
+      mv server "${{targets.destdir}}"/usr/share/${{package.name}}/
+      mv LICENSE.txt "${{targets.destdir}}"/usr/share/licenses/${{package.name}}/
+      cp pom.xml "${{targets.destdir}}"/usr/share/${{package.name}}/
+      cp refine.ini "${{targets.destdir}}"/usr/share/${{package.name}}/
+      cp refine "${{targets.destdir}}"/usr/share/${{package.name}}/
+      install -Dm755 openrefine.sh "${{targets.destdir}}"/usr/bin/openrefine
```

Comment:
Thanks for the contribution!

In Wolfi we prefer to compile software and copy built artifacts into the `${{targets.destdir}}/usr/bin/`.

I'm not familiar with refine but after a quick look I wonder if we can add `openjdk-20` and a compatible version of `nodejs`, maybe `maven` too as I see a pom.xml, to the build environment package list and run
```
./refine build
```
to compile the necessary artifacts and move those to `${{targets.destdir}}/usr/bin/`?

Reply:
I updated the YAML file following your suggestions, @rawlingsj. However, it gets stuck on `Downloading from central: https://repo.maven.apache.org/maven2/org/python/jython-standalone/2.7.3/jython-standalone-2.7.3.jar`. Some file downloads also stay inactive for a while but eventually proceed. I'm running `mvn --debug` to see if it shows any useful debug info, but I have had no luck with it. I guess something on their side blocks after x requests in y time.

I looked at how Debian (Stable branch) packages it, and they include a bunch of dependencies that would be otherwise downloaded. This works for Debian Stable, but I don't know if it's the best approach for a moving project.

I appreciate any feedback and suggestions you may have on how to proceed.

Reply:
OpenRefine has an official Linux distribution of the software, available on the releases page from GitHub. I updated the openrefine.yaml to use that. Is this acceptable?